### ISSUE TITLE
Feature Request - Identity Principal Resource

### ISSUE THREAD

**Issue by adarobin** [vmware/terraform-provider-nsxt#815]
### Is your feature request related to a problem? Please describe.

Recently, I've come across some use cases where it would be useful to be able to create Identity Principals via Terraform.

The use cases are:
- Creating Identity Principals for use with NSX Container Plug-in or VMware Container Networking with Antrea
- Creating Identity Principals to be used by internal "customers" to manage a subset of firewall policy without impacting the entire environment

### Describe the solution you'd like

```terraform
resource "nsxt_identity_principal" "my_principal" {
  name = "my_principal"
  node_id = "asdf"
  certificate_pem = "PEMSTRING"
  role = "enterprise_admin"
}
```

**Comment by salv-orlando** [comment]
@wsquan171 looks like your PR #1021 addresses this issue, is that correct?

**Comment by wsquan171** [comment]
Implemented with #1021
